src/auth-stack.ts

```diff
@@ -359,14 +364,14 @@ export class AuthStack {
       description: 'Auth API JKWS URI',
     })
 
-    new CfnOutput(stack, 'AuthUserPoolId', {
-      value: resourceParams.userPool?.userPoolId!,
-      description: 'Auth UserPoolId',
+    new CfnOutput(stack, 'AuthApiUserPoolId', {
+      value: userPool?.userPoolId!,
+      description: 'Auth API UserPoolId',
     })
 
-    new CfnOutput(stack, 'AuthUserPoolClientId', {
-      value: resourceParams.userPoolClientId!,
-      description: 'Auth UserPoolClientId',
+    new CfnOutput(stack, 'AuthApiUserPoolClientId', {
+      value: userPoolClientId!,
```

Comment:
Why do these `value` fields have a `!` in the end?

Reply:
This is because ResourceParams will be filled at the specific moment in time so that's the reason. If you look at the code we are filling those params along the way and use them in the next steps. In that way we don't pollute all methods with tons of params.